source/config/profileUpgradeSteps.py

```diff
@@ -439,6 +440,31 @@ def upgradeConfigFrom_13_to_14(profile: ConfigObj):
 	log.debug("Deleting old config value.")
 	del profile["speech"]["outputDevice"]
 
+	# """Convert keyboard typing echo configuration from boolean to integer values."""
+	try:
+		speakTypedCharacters = profile["keyboard"]["speakTypedCharacters"]
+		if configobj.validate.is_boolean(speakTypedCharacters):
+			old_value = speakTypedCharacters
+			new_value = TypingEcho.EDIT_CONTROLS.value if speakTypedCharacters else TypingEcho.OFF.value
+			profile["keyboard"]["speakTypedCharacters"] = new_value
```

Comment:
A `KeyError` here (if any) should not be caught. Could you rather use a try/except/else structure, putting in the except clause only one line of code? And same for typed words upgrade below.

Reply:
Regarding the use of the try/except/else structure:  I understand you recommend placing the code that may raise a  KeyError  within the  try  block, keeping only one line in the  except  block to handle the exception, and moving the main logic to the  else  block. Is this correct?

Reply:
Yes that's correct.
Sorry my initial wording was not clear nor completely correct since there is already one line in the except block. I rather mean:
A `KeyError` here (if any) should not be caught. Could you rather use a try/except/else structure, putting in the try block only one line of code? And same for typed words upgrade below.